libflux/src/flux/ast/check/tests.rs

```diff
@@ -57,6 +57,7 @@ fn test_check_collect_existing_error() {
                 source: Some(String::from("a = 1\nb=2\nc=a+b")),
             },
             errors: vec![String::from("error 1")],
+            ..BaseNode::default()
```

Comment:
what is the `..` syntax in Rust? Its new to me.

Reply:
It says populate the remaining, unspecified fields with values from some other node. I would have liked to make more use of this, especially in the testing code, however it only works well if there are simple defaults available. When parse tree nodes are made up of directly contained, or boxed-contained values (non-optional) you have to make a default that essentially cascades into a full parse tree for a valid expression, because the default itself must be complete. I decided to avoid that.

It would be great if rust let you made incomplete defaults that you could then draw on using `..` to fill the remaining fields in from reasonable defaults. As far as I could tell, you can't.

This represents my initial learnings in the language, so maybe there are better ways?